fe/fe-core/src/main/java/com/starrocks/http/rest/TransactionLoadAction.java

```diff
@@ -114,20 +116,105 @@ public class TransactionLoadAction extends RestBaseAction {
+                    .removalListener((key, value, cause) -> {
+                        LOG.debug("Evicted transaction label node mapping: {}->{} ({})", key, value, cause);
+                    })
+                    .recordStats()
+                    .build();
+        }
+
+        public long getMaxCapacity() {
+            long current = (long) GlobalStateMgr.getCurrentState().getNodeMgr().getClusterInfo().getTotalBackendNumber() +
+                    GlobalStateMgr.getCurrentState().getNodeMgr().getClusterInfo().getTotalComputeNodeNumber();
+            return current * 512;
+        }
+
+        public void refreshCapacity() {
+            cache.policy().eviction().ifPresent(eviction -> {
+                if (eviction.getMaximum() != getMaxCapacity()) {
+                    eviction.setMaximum(getMaxCapacity());
+                }
+            });
+        }
+
+        public void put(String key, Long value) {
+            cache.put(key, value);
+            refreshCapacity();
+        }
+
+        public Long get(String key, String dbName, String label) {
+            Long nodeId = cache.getIfPresent(key);
+            if (nodeId == null) {
+                String nodeIp = getTxnCoordinator(dbName, label);
+                if (nodeIp != null) {
+                    if (RunMode.isSharedDataMode()) {
+                        List<ComputeNode> computeNodes =
+                                GlobalStateMgr.getCurrentState().getNodeMgr().getClusterInfo().getComputeNodeOnlyWithHost(nodeIp);
+                        if (computeNodes.size() == 1) {
+                            nodeId = computeNodes.get(0).getId();
+                        } else {
+                            //TODO:Compatible with deploying multiple CNs on the same node
+                            LOG.error("Failed to get compute node id for ip: {}, computeNodes: {}", nodeIp, computeNodes);
+                        }
+                    } else {
+                        List<Backend> backends = GlobalStateMgr.getCurrentState().
+                                getNodeMgr().getClusterInfo().getBackendOnlyWithHost(nodeIp);
+                        if (backends.size() == 1) {
+                            nodeId = backends.get(0).getId();
+                        } else {
+                            //TODO:Compatible with deploying multiple BEs on the same node
+                            LOG.error("Failed to get backend node id for ip: {}, backends: {}", nodeIp, backends);
+                        }
+                    }
+                }
+            }
+
+            return nodeId;
+        }
+
+        public String getTxnCoordinator(String dbName, String label) {
+            GlobalStateMgr globalStateMgr = GlobalStateMgr.getCurrentState();
+            Database db = globalStateMgr.getLocalMetastore().getDb(dbName);
+            if (db == null) {
+                LOG.error("unknown database, database=" + dbName);
```

Comment:
these logs may be useless, it is up to the caller how to deal with the null return value. An alternative way is to throw exception with exception message, and the caller just need to catch the exception and determine whether need to log the exception message.